src/Instances/OnChange.lua

```diff
@@ -19,11 +21,17 @@ local function OnChange(propertyName: string): PubTypes.SpecialKey
 		local ok, event = pcall(applyTo.GetPropertyChangedSignal, applyTo, propertyName)
 		if not ok then
 			logError("cannotConnectChange", nil, applyTo.ClassName, propertyName)
-		elseif typeof(callback) ~= "function" then
+			return
+		end
+		if typeof(callback) ~= "function" and xtypeof(callback) ~= "state" then
 			logError("invalidChangeHandler", nil, propertyName)
 		else
 			table.insert(cleanupTasks, event:Connect(function()
```

Comment:
The event remains connected indefinitely here, regardless of the state's result. This means there is little, if any, performance benefit from the feature. Outside of performance gain, I don't think the feature improves the readability of user code, and I'm stuck wondering if it's a suitable fit for Fusion.

Reply:
It's probably better to talk about issues you have with the idea in the linked issue, rather than mixing it in with code review; I'll drop a comment there with some of my thoughts on the matter.